### PR TITLE
added types to index.d.ts

### PR DESCRIPTION
**Pull request by HIMISOCOOL**
added typescript dev dependency matching vuejs current version.
added leaflet types as a normal dependency as types require it.
scaffolded test folder and command in package.json

It definitely needs checking and testing but most of them worked for my project

### CODE REVIEW

**Comment by lordfuoco** [comment]
@HIMISOCOOL is this ready now ? Shall we merge ? I want to do a release soon :) now I am enabled on npm

**Comment by HIMISOCOOL** [comment]
Yeah, should be good to go, I guess you want to merge this guy after next is in then

**Comment by lordfuoco** [comment]
@HIMISOCOOL next and master are the same :) on next we just release the testing / dev version. Merging this

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 .DS_Store
 node_modules/
 npm-debug.log
-docs/_site/
-dist/
\ No newline at end of file
+docs/_site/
```
Comment (lordfuoco) [inline .gitignore]: can you restore this as per master and remove:

-dist files
-examples/package.json
-package.json version to the one on master

sorry for the inconvenience!

**Pull request merged**